src/main/java/picard/illumina/parser/NewIlluminaDataProvider.java

```diff
@@ -0,0 +1,118 @@
+package picard.illumina.parser;
+
+import picard.illumina.ExtractIlluminaBarcodes;
+import picard.illumina.ExtractIlluminaBarcodes.BarcodeMetric;
+import picard.illumina.parser.readers.AbstractIlluminaPositionFileReader;
+import picard.illumina.parser.readers.CbclReader;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static picard.illumina.ExtractIlluminaBarcodes.PerTileBarcodeExtractor.findBestBarcodeAndUpdateMetrics;
```

Comment:
Agreed but this was a breaking change and it was asked that we don't move it.  We could duplicate the code to a top level class and deprecate this.. but I didn't want to tackle that in this PR.